src/fund/utils/subscribeToWindowMessage.ts

```diff
@@ -0,0 +1,85 @@
+import { DEFAULT_ONRAMP_URL } from '../constants';
+import type { JsonObject } from '../jsonTypes';
+
+export enum MessageCodes {
+  LaunchEmbedded = 'launch_embedded',
+  AppReady = 'app_ready',
+  AppParams = 'app_params',
+  SigninSuccess = 'signin_success',
+  Success = 'success',
+  Exit = 'exit',
+  Event = 'event',
+  Error = 'error',
+  PixelReady = 'pixel_ready',
+  OnAppParamsNonce = 'on_app_params_nonce',
+}
+
+export type MessageCode = `${MessageCodes}`;
+
+export type MessageData = JsonObject;
+
+export type PostMessageData = {
+  eventName: MessageCode;
+  data?: MessageData;
+};
+
+/**
+ * Subscribes to a message from the parent window.
+ * @param messageCode A message code to subscribe to.
+ * @param onMessage Callback for when the message is received.
+ * @param shouldUnsubscribe Whether to unsubscribe after the first message is received.
+ * @param allowedOrigin The origin to allow messages from.
+ * @param onValidateOrigin Callback to validate the origin of the message.
+ * @returns
+ */
+export function subscribeToWindowMessage(
+  messageCode: MessageCode,
```

Comment:
@rustam-cb Seems like the only message code we really care about is event. Maybe we should just delete all the others?

Reply:
@steveviselli-cb 
in onramp-widget we do use `app_params` as well.  If not that then we could just remove the `messageCode` from this function fully. 
https://sourcegraph.cbhq.net/github.cbhq.net/payments/onramp-widget/-/blob/src/pages/embed/index.page.tsx?L22

In this case are you suggesting to create a separate type for the messageCode with two items in it? (`event` and `app_params`)

Reply:
@rustam-cb the whole embeded flow is deprecated. We don't support it any more. The code is still there because customers might still be using old versions of our SDK, but it wont work correctly. Anything to do with embeded onramp should not be included in onchainkit.

I think we can probably remove the messageCode param from the function.